test/coinselector-test.js

```diff
@@ -0,0 +1,335 @@
+        const selection = selector.selectLowestLarger(target);
+
+        let selectedValues = 0;
+        for (const i of selection) {
+          selectedValues += pointers[i].effectiveValue;
+        }
+
+        assert.strictEqual(selectedValues, target);
+      });
+    }
+
+    it('should be able to fund all values in range 1 to 221000 using Lowest Larger', () => {
+      for (let target = 1; target <= 221000; target++) {
+        const selection = selector.selectLowestLarger(target);
+
+        let selectedValues = 0;
+        for (const i of selection) {
+          selectedValues += pointers[i].effectiveValue;
+        }
+
+        assert(selectedValues >= target);
+      }
+    });
+  });
+
+  describe('Single Random Draw Selection', function () {
+    it('should be able to fund all values in range 1 to 221000 using Single Random Draw', () => {
+      for (let target = 1; target <= 221000; target++) {
+        const selection = selector.selectSRD(target);
+
+        let selectedValues = 0;
+        for (const i of selection) {
+          selectedValues += pointers[i].effectiveValue;
+        }
+
+        assert(selectedValues >= target);
+      }
+    });
+  });
+});
+
+const workers = new WorkerPool({
+  enabled: true,
+  size: 2
+});
+
+describe('Integration', function () {
+  this.timeout(1000000);
+
+  const wdb = new WalletDB({workers});
+
+  function fromU32(num) {
+    const data = Buffer.allocUnsafe(4);
+    data.writeUInt32LE(num, 0, true);
+    return data;
+  }
+
+  function nextBlock(wdb) {
+    return fakeBlock(wdb.state.height + 1);
+  }
+
+  function fakeBlock(height) {
+    const prev = hash256.digest(fromU32((height - 1) >>> 0));
+    const hash = hash256.digest(fromU32(height >>> 0));
+    const root = hash256.digest(fromU32((height | 0x80000000) >>> 0));
+
+    return {
+      hash: hash,
+      prevBlock: prev,
+      merkleRoot: root,
+      time: 500000000 + (height * (10 * 60)),
+      bits: 0,
+      nonce: 0,
+      height: height
+    };
+  }
+
+  function dummyInput() {
+    const hash = random.randomBytes(32);
+    return Input.fromOutpoint(new Outpoint(hash, 0));
+  }
+
+  before(async () => {
+    await wdb.open();
+    await workers.open();
+  });
+
+  let oldBalance, newBalance, oldCoins, newCoins;
+
+  after(async () => {
+    console.log('Amount saved in fees :', newBalance - oldBalance);
+    console.log('Coins in UTXO pool using old selection :', oldCoins);
+    console.log('Coins in UTXO pool using new selection :', newCoins);
+    await wdb.close();
+    await workers.close();
+  });
+
+  for (const useSelectEstimate of [true, false]) {
+    it('should send transactions using old selection', async () => {
+      const alice = await wdb.create();
+      const bob = await wdb.create();
+
+      for (let i = 0; i < 6000; i++) {
+        const payment = data[i];
+        let value = Amount.value(payment.value);
+        const rate = Amount.value(payment.rate);
```

Comment:
The `value` and `rate` are in BTC and we need to convert them into sats.

Reply:
The whole data file has over 15000 txs. I hard-coded 6000 because it runs in a reasonable amount of time and shows significant improvement over the old algorithm. If someone wants to play with the test then they can just increase/decrease the limit. I should probably add a comment there regarding this.